Objects/typeobject.c

```diff
@@ -3068,6 +3080,41 @@ PyType_GetSlot(PyTypeObject *type, int slot)
     return  *(void**)(((char*)type) + slotoffsets[slot]);
 }
 
+PyObject *
+PyType_GetModule(PyTypeObject *type) {
+    if (!PyType_Check(type) || !PyType_HasFeature(type, Py_TPFLAGS_HEAPTYPE)) {
```

Comment:
Maybe check PyType_Check(type) with an assertion to reduce the performance overhead at runtime?

I recently converted PyType_HasFeature() to an opaque function. _PyType_HasFeature() static inline function can be used for best performance in typeobject.c:

```suggestion
    assert(PyType_Check(type));
    if (!_PyType_HasFeature(type, Py_TPFLAGS_HEAPTYPE)) {
```